### ISSUE TITLE
nREPL hangs trying to connect to a Luminus project

### ISSUE THREAD

**Issue by yogthos** [LightTable/LightTable#2144]
steps to reproduce

* create a new project with `lein new luminus myapp`
* run the project with `lein run`, this will start nREPL on port `7000`
* open a namespace from the project in LT and try to do add connection nREPL on `localhost:7000`

The `Connecting...` message appears and hangs, no errors appear.

**Comment by cldwalker** [comment]
Hi yogthos. When I try this on osx 10.10, I don't get a hang but instead a connection to a new process (which isn't correct). When I do [a remote connection](https://github.com/LightTable/Clojure#connect-to-remote-nrepl) via `lein repl`, I'm able to correctly communicate with the server process via lein's nrepl server and can run `start-app`. Are you aware of any differences b/n [luminus' nrepl server](https://github.com/luminus-framework/luminus-nrepl/blob/master/src/luminus/repl_server.clj) and [lein's](https://github.com/technomancy/leiningen/blob/master/src/leiningen/repl.clj)? Nothing immediately pops out. I don't have cycles to dig deeper but happy to answer any LT questions if you're interested in looking further into this

**Comment by rundis** [comment]
Beat me to it, I was about to mention that the steps to reproduce didn't take into account the need to add lein-light-nrepl and nrepl-options to project.clj

**Comment by yogthos** [comment]
Yeah, I'm a bit stumped on this one as well. The nrepl started by luminus should be pretty standard and cursive connects just fine, and I can connect by running `lein repl :connect 7000` as well. What's the relationship of lein-light-nrepl and nrepl-options to the app starting the repl though. Wouldn't that be happening on LT end?

**Comment by cldwalker** [comment]
For a given port and host, LT opens a [connection to a nrepl client](https://github.com/LightTable/Clojure/blob/faa4b896bb24d157900508b062a066af04893d24/src/lt/plugins/clojure/nrepl.cljs#L76-L82). The lein-light-nrepl middleware entry fn is [lighttable-ops](https://github.com/LightTable/Clojure/blob/faa4b896bb24d157900508b062a066af04893d24/lein-light-nrepl/src/lighttable/nrepl/handler.clj#L21). Two possible reasons for why the luminus nrepl server has different behavior: different tools.nrepl version or some of the [server options](https://github.com/luminus-framework/luminus-nrepl/blob/766c4c24062e5548e811214f6af7e814202184b3/src/luminus/repl_server.clj#L23-L28) as nil values is causing different behavior than [lein's options](https://github.com/technomancy/leiningen/blob/540142bc55488e757769ca2d1fac1ec3c7253d21/src/leiningen/repl.clj#L173-L175)

**Comment by yogthos** [comment]
I think it's most likely the tools.nrepl version. I tried adding lein-light-nrepl and nrepl-options, but that didn't have any effect. The only option I pass in by default is the [repl port here](https://github.com/luminus-framework/luminus-template/blob/master/resources/leiningen/new/luminus/core/src/core.clj#L28), and I tried adding the same ones as lein does, but that doesn't seem to help either.

**Comment by ashleyconnor** [comment]
Also having this issue.

[Source code](https://github.com/ashleyconnor/guestbook)

Added: `[lein-light-nrepl "0.3.3"]` to `:project/dev :dependencies` and `:repl-options {:nrepl-middleware [lighttable.nrepl.handler/lighttable-ops]}` (see `project.clj` file).

![Screenshot](https://www.dropbox.com/s/hu1cq94clgemict/Screenshot%202016-08-28%2011.11.58.png?dl=1)

**Comment by yogthos** [comment]
For `lein repl`, it should just be `lein repl :connect localhost:7000` without `http://`.

**Comment by alephyud** [comment]
Had the same issue – could finally connect to the nrepl server when I explicitly specified the Light Table handler for nrepl in `repl-server` in the `core` ns of my app:

```
(ns myapp.core
  (:require [luminus.repl-server :as repl]
            [lighttable.nrepl.handler :refer [lighttable-ops]]
            [clojure.tools.nrepl.server :as nrepl]
            ...))

(mount/defstate ^{:on-reload :noop}
                repl-server
                :start
                (when-let [nrepl-port (env :nrepl-port)]
                  (repl/start
                    {:port nrepl-port
                     :handler (nrepl/default-handler #'lighttable-ops)}))
                :stop
                (when repl-server
                  (repl/stop repl-server)))
```

Hope this helps.

**Comment by nenadmitrovic** [comment]
I have the same issue. I tried on Ubuntu and Windows. I got the same issue. After hours of trying to connect it's still trying.

**Comment by prertik** [comment]
> I have the same issue. I tried on Ubuntu and Windows. I got the same issue. After hours of trying to connect it's still trying.
@nenadmitrovic 
https://stackoverflow.com/a/55662215/6587655
